script.module.clouddrive.common/clouddrive/common/export.py

```diff
@@ -0,0 +1,110 @@
+        self._config_path = os.path.join(addon_data_path, self._config_file_name)
+        if not os.path.exists(addon_data_path):
+            try:
+                os.makedirs(addon_data_path)
+            except:
+                KodiUtils.get_system_monitor().waitForAbort(3)
+                os.makedirs(addon_data_path)
+
+    def load(self):
+        self.exports = {}
+        if os.path.exists(self._config_path):
+            with KodiUtils.lock:
+                with open(self._config_path, 'rb') as fo:
+                    self.exports = json.loads(fo.read())
+        return self.exports
+    
+    def add_export(self, export):
+        self.load()
+        self.exports[export['id']] = export
+        self.save()
+    
+    def save(self):
+        with KodiUtils.lock:
+            with open(self._config_path, 'wb') as fo:
+                fo.write(json.dumps(self.exports, sort_keys=True, indent=4))
+
+    def remove_export(self, exportid):
+        self.load()
+        del self.exports[exportid]
+        self.save()
+    
+    def get_items_info_path(self, exportid):
+        return os.path.join(self._addon_data_path, self._export_items_file_format % exportid)
+    
+    def get_items_info(self, exportid):
+        items_info = None
+        items_info_path = self.get_items_info_path(exportid)
+        if KodiUtils.file_exists(items_info_path):
+            with KodiUtils.lock:
+                with open(items_info_path, 'rb') as fo:
+                    items_info = eval(fo.read())
+        return items_info
+    
+    @staticmethod
+    def add_item_info(items_info, item_id, name, full_local_path, parent):
+        items_info[item_id] = {'name': name, 'full_local_path': full_local_path, 'parent': parent}
+        #if parent:
+        #    parents = Utils.get_safe_value(items_info, 'parents', [])
+        #    parents.append(parent)
+        #    items_info['parents'] = parents
+    
+    @staticmethod
+    def remove_item_info(items_info, item_id):
+        if item_id in items_info:
+            #item_info = items_info[item_id]
+            del items_info[item_id]
+            #parent = Utils.get_safe_value(item_info, 'parent')
+            #parents = Utils.get_safe_value(items_info, 'parents')
+            #if parent and parents and parent in parents:
+            #    parents.remove(parent)
+    
+    @staticmethod
+    def create_strm(driveid, item, file_path, content_type, addon_url):
+        item_id = Utils.str(item['id'])
+        item_drive_id = Utils.default(Utils.get_safe_value(item, 'drive_id'), driveid)
+        f = KodiUtils.file(file_path, 'w')
+        f.write(addon_url + '?' + urllib.urlencode({'action':'play', 'content_type': content_type, 'item_driveid': item_drive_id, 'item_id': item_id, 'driveid': driveid}))
+        f.close()
+        return True
+        
+    def save_items_info(self, exportid, items_info):
+        f = KodiUtils.file(self.get_items_info_path(exportid), 'w')
```

Comment:
use `with`

Reply:
Working on it

Reply:
Did it in this way because xbmcvfs.File doesn't have the `__exit__` method where I suppose the file should be closed. 
if I use `with KodiUtils.file(file_path, 'w') as f` I'm getting the exception `AttributeError: __exit__`.
Let me know